### PR TITLE
Clarification of docker section in README.md

### LINKED ISSUE #128: README.md is confusing about data location
Hi,

README says:
> By default the server persists backup data in `/tmp/restic`. To start the server with a custom persistence directory and with authentication disabled

But, at least on docker, the default location is `/data`. This is not mentioned on that sentence neither on the section above about Docker.

Maybe it could be clarified.

thanks,
marc

### PR DESCRIPTION
**Pull request by telenieko**
Fixes #128, the two distinct docker sections of the README are merged into one.

Checklist
---------

- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- ~[ ] I have added tests for all changes in this PR~
- ~[ ] I have added documentation for the changes (in the manual)~
- ~[ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/rest-server/blob/master/changelog/TEMPLATE))~
- ~[ ] I have run `gofmt` on the code in all commits~
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/rest-server/commits/master)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,9 +81,19 @@ There's an example [systemd service file](https://github.com/restic/rest-server/
...
+    docker build -t restic/rest-server:latest .
+
+Otherwise just pull the current release directly:
+
+    docker pull restic/rest-server:latest
+
 By default, image uses authentication.  To turn it off, set environment variable `DISABLE_AUTHENTICATION` to any value.
 
-Persistent data volume is located to `/data`.
+Note that, **contraty to the defaults** of `restic-server`, the persistent data volume is located to `/data`.
```
Comment (Enrico204) [inline README.md:96]: "contrary" is misspelled :-)

However I think that the original formulation is enough (I mean "Persistent data volume is located to `/data`). Maybe the whole original phase might be written in bold. However that's only my opinion :-D

**Review by Enrico204 — CHANGES_REQUESTED** [review]

**Review by telenieko — COMMENTED** [review]
Fixed typo

File: README.md (hunk 1 above)
Comment (telenieko) [reply README.md]: > "contrary" is misspelled :-)

fixed.

> However I think that the original formulation is enough (I mean "Persistent data volume is located to `/data`). Maybe the whole original phase might be written in bold. However that's only my opinion :-D

I wanted to emphasize that the image's behaviour differs from the defaults above. I believe it to be a good practice to make it clear when something overrides a default!

File: README.md [hunk 2]
```diff
@@ -99,9 +81,19 @@ There's an example [systemd service file](https://github.com/restic/rest-server/
...
+
+This repository hold the `Dockerfile` of such image, so you may build a fresh one from git via:
+
+    docker build -t restic/rest-server:latest .
+
+Otherwise just pull the current release directly:
+
+    docker pull restic/rest-server:latest
+
 By default, image uses authentication.  To turn it off, set environment variable `DISABLE_AUTHENTICATION` to any value.
```
Comment (MichaelEischer) [inline README.md:94]: I think this and the next line should be moved to the `Start server` section.

File: README.md [hunk 3]
```diff
@@ -99,9 +81,19 @@ There's an example [systemd service file](https://github.com/restic/rest-server/
...
 
-Persistent data volume is located to `/data`.
+Note that, **contrary to the defaults** of `restic-server`, the persistent data volume is located to `/data`.
```
Comment (MichaelEischer) [inline README.md:96]: s/restic-server/rest-server/

File: README.md [hunk 4]
```diff
@@ -99,9 +81,19 @@ There's an example [systemd service file](https://github.com/restic/rest-server/
 
 ### Docker
 
+Restic works well inside a container, images are [published to Docker Hub](https://hub.docker.com/r/restic/rest-server).
```
[the remark this answers is not in the document]
Comment (MichaelEischer) [reply README.md:84]: As the Docker Hub is referenced here, the remainder of this section should first describe how to pull the image and then how to build a custom one.

File: README.md [hunk 5]
```diff
@@ -99,9 +81,19 @@ There's an example [systemd service file](https://github.com/restic/rest-server/
...
+
+This repository hold the `Dockerfile` of such image, so you may build a fresh one from git via:
```
[the remark this answers is not in the document]
Comment (telenieko) [reply README.md]: The wording of that paragraph has now changed

**Comment by telenieko** [comment]
@MichaelEischer previous comments have been integrated into the proposed text, also integrated the merge conflict with the master branch.

If/When merging please squash commits to not fill master with the small changes

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks for the cleanups!

**Pull request merged**